ui/components/multichain-accounts/address-qr-code-modal/address-qr-code-modal.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+import qrCode from 'qrcode-generator';
+import {
+  Text,
+  TextVariant,
+  TextAlign,
+  TextColor,
+  Button,
+  IconName,
+  ButtonVariant,
+  ButtonSize,
+  Box,
+  BoxFlexDirection,
+  BoxJustifyContent,
+  BoxAlignItems,
+  AvatarNetwork,
+  FontWeight,
+} from '@metamask/design-system-react';
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+} from '../../component-library';
+import type { ModalProps } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import { openBlockExplorer } from '../../multichain/menu-items/view-explorer-menu-item';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+
+// Constants for QR code generation
+const QR_CODE_TYPE_NUMBER = 4;
+const QR_CODE_CELL_SIZE = 5;
+const QR_CODE_MARGIN = 16;
+const QR_CODE_ERROR_CORRECTION_LEVEL = 'M';
+
+// Constants for address segmentation
+const PREFIX_LEN = 6;
+const SUFFIX_LEN = 5;
+
+export type AddressQRCodeModalProps = Omit<
+  ModalProps,
+  'isOpen' | 'onClose' | 'children'
+> & {
+  isOpen: boolean;
+  onClose: () => void;
+  address: string;
+  accountName: string;
+  networkName: string;
+  networkImageSrc?: string | undefined;
+};
+
+export const AddressQRCodeModal: React.FC<AddressQRCodeModalProps> = ({
+  isOpen,
+  onClose,
+  address,
+  accountName,
+  networkName,
+  networkImageSrc,
+}) => {
+  const t = useI18nContext();
+  const [copied, handleCopy] = useCopyToClipboard();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  // Address segmentation for display
+  const addressStart = address.substring(0, PREFIX_LEN);
+  const addressMiddle = address.substring(
+    PREFIX_LEN,
+    address.length - SUFFIX_LEN,
+  );
+  const addressEnd = address.substring(address.length - SUFFIX_LEN);
```

Comment:
Setting up address segmentation to render the address in different colors 

<img width="276" height="54" alt="Screenshot 2025-09-03 at 4 37 33 PM" src="https://github.com/user-attachments/assets/673a86a8-0aff-4695-a556-a661c2a0cc82" />